plugins/modules/global_accelerator.py

```diff
@@ -0,0 +1,411 @@
+      - Two IPs maximum
+      - If not supplied, AWS will provision IPs for you
+      - If you only supply one, AWS will provision a second one for you
+      - See U(https://docs.aws.amazon.com/global-accelerator/latest/dg/using-byoip.html) for more information
+    type: list
+  name:
+    description:
+      - The name of the Accelerator
+      - If performing an update and more than one accelerator exists with this name, the module will abort and fail
+      - Mutually exclusive with I(arn)
+    type: str
+  purge_tags:
+    description:
+      - If yes, existing tags will be purged from the resource to match exactly what is defined by I(tags) parameter.
+      - If the I(tags) parameter is not set then tags will not be modified.
+    type: bool
+  region:
+    description:
+      - Global Accelerators must currently be created and modified in us-west-2
+      - Global Accelerators still operate across all regions
+    type: str
+  state: 
+    description:
+      - Create or destroy the global accelerator
+      - I(enabled) must be false to destroy an accelerator
+    type: str
+  tags:
+    description:
+      - A dictionary of one or more tags to assign to the load balancer.
+    type: dict
+"""
+
+EXAMPLES = """
+# Note: These examples do not set authentication details, see the AWS Guide for details
+
+# Create a Global Accelerator
+- global_accelerator:
+    enabled: true
+    name: myga
+    state: present
+
+# Create a Global Accelerator with BYOIP
+- global_accelerator:
+    enabled: true
+    name: myga
+    ip_addresses:
+      - 1.2.3.4
+      - 5.6.7.8
+"""
+
+
+def get_accelerator(connection, module, name, arn):
+    accelerator = None
+
+    if name and not arn:
+        module.debug("Looking up Accelerators with Name {}".format(name))
+        # There is no direct way to look up Accelerators by ARN
+        # So we fetch them all and do client side filtering
+        try:
+            paginator = connection.get_paginator('list_accelerators')
+
+            accelerators = AWSRetry.jittered_backoff()(paginator.paginate)().build_full_result()['Accelerators']
+
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e)
+
+        # Filter all results for the specified name
+        accelerators = [a for a in accelerators if a['Name'] == name]
+
+        if len(accelerators) > 1:
+            module.fail_json(
+                msg="More then one accelerator found with name {}. Refusing to act".format(name)
+            )
+
+        if not accelerators:
+            module.warn("No accelerator found with name {}".format(name))
+        else:
+            accelerator = accelerators[0]
+
+    if arn:
+        accelerator = AWSRetry.jittered_backoff()(
+            connection.describe_accelerator
+        )(AcceleratorArn=arn)['Accelerator']
+
+    if accelerator:
+        accelerator = camel_dict_to_snake_dict(accelerator)
+
+    return accelerator
+
+
+class GlobalAccelerator(object):
```

Comment:
TBH I'm mad confused about where the various files for this repo live. Based on other modules here, it seems like this class should live somewhere else. I'm happy to move it if someone can point me in a direction

Reply:
Is this class needed for this module?  These methods should be able to be moved into the module and called from main() based on the action the user is requesting.